### PR TITLE
Java: Update MaD Declarations after Triage

### PR DESCRIPTION
**Pull request by kaeluka**
This PR was created by @kaeluka.

The PR was done in pairing with @atorralba and @jcogs33.

TR 2324

### CODE REVIEW

**Comment by kaeluka** [comment]
@jcogs33 @atorralba Hey there! Thanks for today — this is the PR I promised. I also started a DCA experiment (using the queries: nightly, repos: nightly). It might be a good idea for one of you to double check that I got the DCA settings right to avoid blocking on it later :)

**Comment by atorralba** [comment]
DCA configuration looks good! As soon as we get results back and we review any potential new alerts, we'll be ready to merge this 🚀

File: java/ql/lib/ext/hudson.model.model.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+      extensible: summaryModel
+    data:
+      - ["hudson.model", "DirectoryBrowserSupport$Path", False, "Path", "(String,String,boolean,long,boolean,long)", "", "Argument[0]", "Argument[-1].SyntheticField[hudson.model.DirectoryBrowserSupport$Path.href]", "taint", "ai-generated"]
+  - addsTo:
+      pack: codeql/java-all
+      extensible: sinkModel
+    data:
+      - ["hudson.model", "DirectoryBrowserSupport", False, "DirectoryBrowserSupport", "(ModelObject,FilePath,String,String,boolean)", "", "Argument[1]", "read-file", "ai-generated"]
+      - ["hudson.model", "UpdateCenter$UpdateCenterConfiguration", True, "download", "(DownloadJob,URL)", "", "Argument[1]", "open-url", "ai-generated"]
+# suggested label is not supported:      - ["hudson.model", "UpdateCenter$UpdateCenterConfiguration", True, "install", "(DownloadJob,File,File)", "", "Argument[1]", "delete-file", "ai-generated"]
```
Comment (atorralba) [inline java/ql/lib/ext/hudson.model.model.yml:13]:
```suggestion
      - ["hudson.model", "UpdateCenter$UpdateCenterConfiguration", True, "install", "(DownloadJob,File,File)", "", "Argument[1]", "create-file", "ai-generated"] # should be delete-file
```

File: java/ql/lib/ext/hudson.util.jna.model.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+extensions:
+  - addsTo:
+      pack: codeql/java-all
+      extensible: sinkModel
+    data:
+      - ["hudson.util.jna", "GNUCLibrary", True, "open", "(String,int)", "", "Argument[0]", "read-file", "ai-generated"]
+# suggested label is not supported:      - ["hudson.util.jna", "Kernel32", True, "MoveFileExA", "(String,String,int)", "", "Argument[0]", "delete-file", "ai-generated"]
```
Comment (atorralba) [inline java/ql/lib/ext/hudson.util.jna.model.yml:7]:
```suggestion
      - ["hudson.util.jna", "Kernel32", True, "MoveFileExA", "(String,String,int)", "", "Argument[0]", "delete-file", "ai-generated"] # should be delete-file
```

**Review by atorralba — APPROVED** [review]
DCA looks uneventful, so let's go ahead and merge once CI is green! 😄

File: java/ql/lib/ext/hudson.util.jna.model.yml (hunk 2 above)
Comment (kaeluka) [reply java/ql/lib/ext/hudson.util.jna.model.yml]: 🙈 This suggestion was incomplete -- it left the `delete-file` label in place. I also didn't spot this. Fixed now.

Comment (atorralba) [reply java/ql/lib/ext/hudson.util.jna.model.yml]: Sorry 🤦. Thanks for fixing it!

**Review by atorralba — APPROVED** [review]

**Pull request merged**